### PR TITLE
Feature/annotation tests

### PR DESCRIPTION
**Pull request by Marcel-Mueck**
# What

The changes for this PR are threefold
- added tests for each function that is called from the annotaton pipeline
- changed some details about the annotation pipeline as well as the environment that it uses
- solved the ID problem described in #81 

# Testing

Testing is performed by github actions

### CODE REVIEW

File: tests/annotations/test_annotations.py [hunk 1]
```diff
@@ -0,0 +1,642 @@
...
+    result = cli_runner.invoke(annotations_cli, cli_parameters, catch_exceptions=False)
+    assert result.exit_code == 0
+    written_results = pd.read_parquet(output_path)
+    expected_results = pd.read_parquet(expected_path)
+    assert written_results.shape == expected_results.shape
+    assert_frame_equal(written_results, expected_results[written_results.columns], check_exact = False)
+
+
+
+# @pytest.mark.parametrize(
```
Comment (endast) [inline tests/annotations/test_annotations.py:610]: Not used?

File: pipelines/resources/absplice.yaml [hunk 2]
```diff
@@ -29,3 +29,4 @@ dependencies:
     - interpret == 0.2.7
     - interpret-core == 0.2.7
     - git+https://github.com/gagneurlab/splicemap.git@9e9831f32c221e850e26757a5f1c132dcd565640
+    - git+https://github.com/gagneurlab/absplice.git
```
Comment (endast) [inline pipelines/resources/absplice.yaml:32]: Can we lock this one as well?

**Review by endast — COMMENTED** [review]
Nice, i just saw a few small things that we might want to address.

File: pipelines/resources/absplice_download.snakefile [hunk 3]
```diff
@@ -124,9 +124,10 @@ if absplice_main_conf['use_rocksdb'] == True:
         params:
             version = genome_mapper[absplice_main_conf['genome']]
         conda:
-            f"./environment_spliceai_rocksdb.yaml"
+            f"./environment_spliceai_rocksdb2.yaml"
+            #f"./environment_spliceai_rocksdb.yaml"
```
Comment (endast) [inline pipelines/resources/absplice_download.snakefile:128]: Not used?

File: pipelines/resources/absplice.yaml (hunk 2 above)
Comment (Marcel-Mueck) [reply pipelines/resources/absplice.yaml]: done

File: pipelines/resources/absplice_splicing_pred_DNA.snakefile [hunk 4]
```diff
@@ -71,7 +71,7 @@ if absplice_main_conf['use_rocksdb'] == True:
             lookup_only = False,
             genome = genome_mapper[absplice_main_conf['genome']]
         conda:
-            f"./environment_spliceai_rocksdb.yaml"
+            f"./environment_spliceai_rocksdb2.yaml"
```
[the remark this answers is not in the document]
Comment (Marcel-Mueck) [reply pipelines/resources/absplice_splicing_pred_DNA.snakefile]: good catch, changed to original env

File: tests/annotations/test_annotations.py (hunk 1 above)
Comment (endast) [reply tests/annotations/test_annotations.py:655]: The whole test is commented out 
image

**Review by endast — APPROVED** [review]
Looks good! 
image

**Pull request merged**